### DISCUSSION TITLE
Hyperspi

### DISCUSSION THREAD

**Discussion by jt7005** [awawa-dev/HyperSPI#21] [General]
Hi, I want to try the Multi-Segment Wiring method on my TV
 Using a Raspberry Pi 3 and a Wemos D1 mini pro (CP2104)
 as the driver and a 74AHCT125 level shifter as per your picture 
 I have two questions, what pins go from the pi (pi3)to what pins of the wemos
 second question is it worth changing as everything seems fine with my current setup, there is a slight delay (reaction time from led)not much
 
 My current setup 
 Raspberry Pi 3 to a ESP8266
 ESP8266 to 250 led sk6812 RGBW
 no level shifter
 Hyperhdr 19
 Hyperspi 9

**Comment by awawa-dev** [comment]
Hi
Unfortunately multi-segment is not supported by esp8266: board hardware limitation. I could take a look into your HyperHDR logs (taken after ~3 minutes after startup) if you upload it on pastebin.com or here. Your setup is rather small, multi-segment would improve things only slightly.

**Reply by jt7005** [reply]
Thanks for the fast reply... What if an added an esp-wroom32 and a 74AHCT125 level shifter, would that improve my current setup?

**Reply by awawa-dev** [reply]
It is difficult to assess the situations without logs. Level shifter will not improve performance.

**Reply by jt7005** [reply]
Hi,
    please find my log:  https://pastebin.com/qRckrgBS

I love Hyperhdr, thank you for all your amazing work

**Reply by awawa-dev** [reply]
Thanks :)
Please set your grabber capturing resolution to  640x480@60fps mode and enable the 'Quarter of frame mode' in the grabber properties.
You have only 10FPS currently:
```
2023-03-06T02:59:19.356Z [V4L2:AUTO] *************************************************************************************************
2023-03-06T02:59:19.356Z [V4L2:AUTO] Starting V4L2 grabber. Selected: USB3.0 UHD: USB2.0 HD (video0) [/dev/video0] 640 x 480 @ 10 fps mjpeg
2023-03-06T02:59:19.356Z [V4L2:AUTO] *************************************************************************************************
```
and the processing time is high 16ms (Quarter of frame should reduce it):
```
2023-03-06T03:02:31.236Z [PERFORMANCE] [USB capturing: FPS = 10.01, decoding = 16ms, frames = 601, invalid = 0], [INSTANCE0: FPS = 10.01, processed = 601], [LED0: FPS = 50.00, send = 3000, processed = 3000, dropped = 0]
```

**Reply by jt7005** [reply]
Wow, its much quicker now , thank you so much
[PERFORMANCE] [USB capturing: FPS = 60.00, decoding = 6ms, frames = 3600, invalid = 0], [INSTANCE0: FPS = 60.00, processed = 3601], [LED0: FPS = 50.00, send = 3000, processed = 3000, dropped = 0]
